articles/matlab-image-database/index.md

```diff
@@ -0,0 +1,133 @@
+
+This is how it is created;
+
+We first initialize the webcam.
+
+```Matlab
+cam = webcam;
+```
+We then enter the required number of the faces to make our database. It is executed by the code below;
+
+```Matlab
+nof = input('enter no. of required frames:');
+```
+We input the number of required frames, for example, if you require 100 or 20 images, you can enter the number of required frames here;
+
+```Matlab
+count = 1;
+```
+We then initialize the count to enter in the while loop. When the count is less than or equal to the number of the required frame, then the while loop will be executed and the snapshot is captured. The captured image is stored into the `img` and this is passed to the `myFaceDetect` function.
+This function should be defined in a different script and then called into the face database script. The description of `myFaceDetect` function is given at the end of the program.
+
+```Matlab
+while count <=nof
+img = snapshot(cam);
+```
+The code returns a cropped image. This means that the function for the cropped image will be used here.
+
+```Matlab 
+[croppedFrame, bboxpoint] = myFaceDetect(img);
+imshow(croppedFrame)
+```
+Note that, in the code above we are not interested in the bounding boxes but the frame. The cropped frame will be stored in the same directory.
+We will then use the `if` and `else` statements to introduce the conditions for running the program.
+If the cropped frame is not empty, the face is detected and we are getting a craft frame, then you can store this frame into any of your folders. It is preferable to create a face database folder in your system for easy identification. We then give the path of the folder in the directory.
+
+```Matlab
+if ~isempty(croppedFrame)
+fileName = strcat(‘directory’,sprintf(‘%d.png’,count));
+```
+We then use the `imwrite` function to write the cropped frame with the filename which is provided in the loop.
+
+```Matlab
+imwrite(croppedFrame,’fileName’);
+msg = [‘imageAcquiredNo:’,num2str(count)]
+disp(msg)
+count = count + 1;
+```
+
+![this is a sample image](/engineering-education/matlab-image-database/image_one.png)
+
+The `num2str` function provides the filename. This means that since the count is starting from one up to ‘n’ which is the no. of given frames, so filenames will be in the form of numbers.
+In the filename above, we gave the file format to the `.png`. This means that we expect the name to be in the form below;
+imageDatabase1.png
+imageDatabase2.png
+imageDatabase3.png until
+imageDatabasen.png  
+
+![this is how the image database appears](/engineering-education/matlab-image-database/database.png)
+
+If there is no face detected in the cropped frame, then it should display `no face detected` in the workspace. Here is the code;
+
+```Matlab
+else
+disp(‘no face detected’)
+```
```

Comment:
Since you have written `if` in separate part and `else` in separate part, there are chances that the reader might get confused. So, additionally have another code snippet containing the full `if-else` part together.